Gems/Atom/RPI/Code/Include/Atom/RPI.Reflect/Material/MaterialAsset.h

```diff
@@ -94,17 +94,20 @@ namespace AZ
 
             //! Returns the shader resource group layout that has per-object frequency. What constitutes an "object" is an
             //! agreement between the FeatureProcessor and the shaders, but an example might be world-transform for a model.
-            //! All shaders in a material will have the same per-object SRG layout.
+            //! All shaders with the same DrawItemType in a material will have the same per-object SRG layout.
             //! @param supervariantIndex: supervariant index to get the layout from.
-            const RHI::Ptr<RHI::ShaderResourceGroupLayout>& GetObjectSrgLayout(const SupervariantIndex& supervariantIndex) const;
+            const RHI::Ptr<RHI::ShaderResourceGroupLayout>& GetObjectSrgLayout(
+                const ShaderCollection::Item::DrawItemType drawItemType, const SupervariantIndex& supervariantIndex) const;
 
             //! Same as above but accepts the supervariant name. There's a minor penalty when using this function
             //! because it will discover the index from the name.
-            const RHI::Ptr<RHI::ShaderResourceGroupLayout>& GetObjectSrgLayout(const AZ::Name& supervariantName) const;
+            const RHI::Ptr<RHI::ShaderResourceGroupLayout>& GetObjectSrgLayout(
+                const ShaderCollection::Item::DrawItemType drawItemType, const AZ::Name& supervariantName) const;
```

Comment:
This (and some other methods in e.g. `MaterialTypeAsset.h`) does change the API. You could move the new parameter to the end and default initialize it with Raster if you want to keep the API "unchanged" for existing code. I have no hard opinion on this as I don't know if this is used anywhere else. If it possibly is, we could still leave it and do a post in impactful changes.

Reply:
moved the parameter to the end and added a "Raster" as default.